dallinger/experiment_server/dashboard.py

```diff
@@ -652,6 +661,229 @@ def dashboard_lifecycle():
     )
 
 
+def clean_http_request_log(msg: str) -> str:
+    """Remove redundant prefix from HTTP request log messages.
+
+    Parameters
+    ----------
+    msg : str
+        The log message to clean
+
+    Returns
+    -------
+    str
+        The cleaned message with redundant prefix removed
+
+    Examples
+    --------
+    >>> msg = '127.0.0.1 - - [20/Mar/2024:10:00:00] "GET /dashboard HTTP/1.1" 200 -'
+    >>> clean_http_request_log(msg)
+    'GET /dashboard HTTP/1.1" 200 -'
+    """
+    is_http_request = "GET " in msg or " POST" in msg
+    has_expected_ending = msg.endswith(("-", '"'))
+
+    if is_http_request and has_expected_ending:
+        return '"'.join(msg.split('"')[1:])
+    return msg
+
+
+def clean_line_info(line_info: dict, log_line_number: int = None) -> dict:
+    """
+    Clean the line info for the log viewer. Convert ANSI to HTML (e.g. ANSI bold to HTML bold) and remove the leading
+    timestamp and log level (because it is already displayed in the table) and add the log's line number if provided.
+
+    :param line_info: The line info to clean
+    :type line_info: dict
+
+    :param log_line_number: The line number in the log file (optional)
+    :type log_line_number: int
+
+    :return: The cleaned line info
+    """
+    msg = line_info["message"]
+    msg = clean_http_request_log(msg)
+    msg = Ansi2HTMLConverter().convert(msg)
+    parsed_msg = bs4.BeautifulSoup(msg, "html.parser")
+    msg = f"<html>{parsed_msg.head}{parsed_msg.body}</html>"
+    line_info["message"] = msg
+    if log_line_number is not None:
+        line_info["log_line_number"] = log_line_number
+    return line_info
+
+
+LOG_FILE = get_logger_filename()
+
+
+def log_read_lines(line_start: int, line_end: int) -> tuple[list[dict], bool, int]:
+    """
+    Read the log file and return the lines in the specified range.
+
+    :param line_start: The line number to start reading from
+    :type line_start: int
+
+    :param line_end: The line number to stop reading at
+    :type line_end: int
+
+    :return: A tuple containing the lines, a boolean indicating if the end of the file was reached, and the last line number
+
+    :Note: The line numbers are 1-based
+
+    """
+    lines = []
+    line_range = range(line_start, line_end)
+    early_stop = False
+    with open(LOG_FILE) as f:
+        for i, line in enumerate(f):
+            number = i + 1
+            if number in line_range:
+                line_dict = clean_line_info(json.loads(line), number)
```

Comment:
Consider renaming `line_dict` as `line_info` and `clean_line_info` as `get_line_info` and moving `json.loads` into `get_line_info` (but I don't really mind)